sql/core/src/test/scala/org/apache/spark/sql/SparkSessionBuilderSuite.scala

```diff
@@ -123,4 +126,95 @@ class SparkSessionBuilderSuite extends SparkFunSuite {
       session.stop()
     }
   }
+
+  test("fork new session and inherit a copy of the session state") {
+    val activeSession = SparkSession.builder().master("local").getOrCreate()
+    val forkedSession = activeSession.cloneSession()
+
+    assert(forkedSession ne activeSession)
+    assert(forkedSession.sessionState ne activeSession.sessionState)
+
+    forkedSession.stop()
+    activeSession.stop()
+  }
+
+  test("fork new session and inherit sql config options") {
```

Comment:
For each other these state, you have to test the following. 

- whether the state gets copied over and is usable
- whether the original session is unaffected by changes to clones session, AND vice versa.

I think the last one is missing in these tests.